server/api/src/rsFileChksum.cpp

```diff
@@ -185,7 +188,37 @@ int fileChksum(rsComm_t* rsComm,
         return error.code();
     }
 
-    auto bytes_remaining = irods::experimental::replica::replica_size(*rsComm, objPath, leaf_resc);
+    std::size_t bytes_remaining;
+
+    namespace ir  = irods::experimental::replica;
+
+    // If we're dealing with an intermediate replica, the data size may not be in the catalog just yet.
```

Comment:
we shouldn't be able to calculate/read an intermediate replica at all?

Reply:
I'll have to check if that is enforced.

This block handles the case around `iput -K`. The checksum api is called before the data size is recorded in the catalog. To handle that, we record the size in the replica state table and the checksum api retrieves the size from the table if the replica is in an intermediate state.

Reply:
right - okay - if it's **our** intermediate replica, then that seems okay.

Reply:
Due to changes in the implementation, this conversation no longer applies.